loki/transformations/hoist_variables.py

```diff
@@ -204,12 +205,17 @@ class HoistVariablesTransformation(Transformation):
     ----------
     as_kwarguments : boolean
         Whether to pass the hoisted arguments as `args` or `kwargs`.
+    remap_dimensions : boolean
```

Comment:
Is there a situation where we wouldn't want to do this? It seems to me like we might not need to make this an option and instead always do this.

Reply:
Good question ... I can't think of a situation we don't want to do that but on the other hand I think having a switch is not a bad idea?!
Anyway, I switched the default but if you insist we can remove this flag.

Reply:
No, flipping the default should be enough. Thanks!